### PR TITLE
feat: Add support for Giphy API's in lexical

### PR DESCRIPTION
**Pull request by rimildeyjsr**
**Changelog:**
1. Added `@giphy/react-components, @giphy/js-fetch-api, @giphy/js-types, react-use` to the lexical-playground package
2. Remove the hard-coded cat gif
3. Added a component to display the giphy plugin in the modal
4. Insert a gif on select

Relates to #1820 

**Screen recording**

https://user-images.githubusercontent.com/10229595/176993308-a619ceaa-c237-4f52-9f4f-b8e7f6a9ad24.mp4

### CODE REVIEW

**Comment by rimildeyjsr** [comment]
@trueadm @zurfyx - While testing locally, I noticed that the GIF is not getting seen on the editor, but the node has been added. Since the GIF is using an image node, I tried to insert an image using an Unplash URL, which has the same result. I don't know if this is a bug or not.

Attached is a screen recording for reference -

https://user-images.githubusercontent.com/10229595/168484624-ee8c8b91-3adb-4595-8044-77a4e0d87d51.mov

**Comment by zurfyx** [comment]
That's looking awesome! Thank you for working on this.

> While testing locally, I noticed that the GIF is not getting seen on the editor, but the node has been added.

The page URL itself won't work, it's not a renderable image, just a page link. You have to point it to a specific image. For example:
```
onSelect({
          altText: item.title,
          src: item.images.original.url,
        })
```

You may want to read over the `item` payload, there's quite a lot of interesting data in there!

File: packages/lexical-playground/src/plugins/ToolbarPlugin.tsx [hunk 1]
```diff
@@ -443,6 +439,36 @@ function InsertImageDialog({
...
+  onClose: () => void;
+}): JSX.Element {
+  const GIPHY_API_KEY = 'gslE624Ax1m7LThTdDztLGZpci6XhfIL';
+  const onSelect = (payload: InsertImagePayload) => {
+    activeEditor.dispatchCommand(INSERT_IMAGE_COMMAND, payload);
+    onClose();
+  };
+
+  return (
+    <ReactGiphySearchbox
```
Comment (zurfyx) [inline packages/lexical-playground/src/plugins/ToolbarPlugin.tsx:456]: Can we use the official Giphy components please? https://github.com/Giphy/giphy-js/tree/master/packages/react-components

From their README:
```
import { Grid } from '@giphy/react-components'
import { GiphyFetch } from '@giphy/js-fetch-api'
// use @giphy/js-fetch-api to fetch gifs
// apply for a new Web SDK key. Use a separate key for every platform (Android, iOS, Web)
const gf = new GiphyFetch('your Web SDK key')

const searchTerm = 'dogs'
// fetch 10 gifs at a time as the user scrolls (offset is handled by the grid)
// if this function changes, change the Grid key to recreate the grid and start over
// see the codesandbox for a runnable example
const fetchGifs = (offset: number) => gf.search(searchTerm, { offset, limit: 10 })
// React Component
ReactDOM.render(<Grid width={800} columns={3} gutter={6} fetchGifs={fetchGifs} key={searchTerm} />, target)
```

Comment (rimildeyjsr) [reply packages/lexical-playground/src/plugins/ToolbarPlugin.tsx:456]: @zurfyx - i will work on this and get back, thanks for the review! 😄

**Comment by im-adithya** [comment]
Hey @rimildeyjsr are you working on this?

**Comment by rimildeyjsr** [comment]
@im-adithya - I am!

**Comment by rimildeyjsr** [comment]
@zurfyx - Hey I am still working on this, sorry for the delay! Hit a snag with the giphy react components, I will try to resolve and get this to a close! :)
Sorry for the delay!

**Comment by rimildeyjsr** [comment]
Opened an issue here for giphy react components, once I get pointers I will resume work here
https://github.com/Giphy/giphy-js/issues/301

**Comment by rimildeyjsr** [comment]
@zurfyx @trueadm - I am done with the integration, let me know what you think!

**Comment by acywatson** [comment]
Thanks for this PR! Looks like we have some issues with the tests. Can you try rebasing and resolving the conflicts? In the meantime, I will look at the code :)

**Comment by rimildeyjsr** [comment]
@acywatson - I have made the changes to package-lock.json, should be fine now!

**Comment by acywatson** [comment]
> @acywatson - I have made the changes to package-lock.json, should be fine now!

Thanks - I think the package lock is still causing issues in the vercel deployment. Can you please try:

delete package-lock.json locally
run npm install
commit the new generated package-lock.json

That should fix it.

**Comment by rimildeyjsr** [comment]
@acywatson - I deleted the package-lock JSON again in local and ran npm install and committed after rebasing, but still, the vercel deployment is failing. Is there any issue on the deployment side? The PR is rebased against main as well

**Comment by acywatson** [comment]
> 

Oh can you try moving these dependencies to the package.json in packages/lexical-playground? Make sure to regenerate the package-lock again after. Sorry for all the back and forth.

**Comment by rimildeyjsr** [comment]
@acywatson - followed the steps and moved dependencies to packages/lexical-playground and regenerated the package-lock.json but I can see that the lexical playground deployment is still failing. Anything else I need to do here? The branch is rebased against main

**Comment by thegreatercurve** [comment]
Closing this as the diff as it is pretty stale now. On reflection, I think we should close the related issue as well. Since it was created, we've added a lot of additional functionality to the playground, and I'm concern that the UX is becoming unwieldy and overwhelming for newcomers. Adding in Giphy doesn't necessarily show how to do anything new that isn't in another rich text media example, like image, Excalidraw, YoutTube or Twitter nodes.

**Pull request closed without merging**